pleasew

```diff
@@ -1,99 +1,122 @@
-#!/usr/bin/env bash
+#!/bin/sh
+
+set -e
 set -u
 
-RED="\x1B[31m"
-GREEN="\x1B[32m"
-YELLOW="\x1B[33m"
-RESET="\x1B[0m"
+set -o errexit
+
+RED='\x1B[31m'
+GREEN='\x1B[32m'
+YELLOW='\x1B[33m'
+RESET='\x1B[0m'
 
-DEFAULT_URL_BASE="https://get.please.build"
+DEFAULT_URL_BASE='https://get.please.build'
 
 OS="$(uname)"
 
-if [[ $OS == "Darwin" ]]; then
-    # switch between mac amd64/arm64 
+if [ "${OS}" = 'Darwin' ]; then
+    # switch between mac amd64/arm64
     ARCH="$(uname -m)"
 else
     # default to amd64 on other operating systems
     # because we only build intel binaries
-    ARCH="amd64"
+    ARCH='amd64'
 fi
 
-# Check PLZ_CONFIG_PROFILE or fall back to arguments for a profile.
-PROFILE="${PLZ_CONFIG_PROFILE:-$(sed -E 's/.*--profile[= ]([^ ]+).*/\1/g' <<< "$*")}"
+# Check `PLZ_CONFIG_PROFILE` or fall back to arguments for a profile.
+PROFILE="${PLZ_CONFIG_PROFILE:-$(
+	sed -E 's/.*--profile[= ]([^ ]+).*/\1/g' <<-EOS
+		"${*}"
+	EOS
+)}"
 
 # Config files on order of precedence high to low.
-CONFIGS=(
-  ".plzconfig.local"
-  "${PROFILE:+.plzconfig.$PROFILE}"
-  ".plzconfig_${OS}_${ARCH}"
-  ".plzconfig"
-  "$HOME/.config/please/plzconfig"
-  "/etc/please/plzconfig"
-)
-
-function read_config() {
-   grep -i "$1" "${CONFIGS[@]}" 2>/dev/null | head -n 1
+CONFIGS="$(cat <<- EOS
+	.plzconfig.local
+	${PROFILE:+.plzconfig.${PROFILE}}
+	.plzconfig_${OS}_${ARCH}
+	.plzconfig
+	${HOME}/.config/please/plzconfig
+	/etc/please/plzconfig
+EOS
+)"
+
+read_config() {
+    printf "%s" "${CONFIGS}" | xargs grep -i "${1}" 2>/dev/null | head -n 1
```

Comment:
This is going to grep over the file name, not the contents of the file unfortunately. We might just want a go old fashioned bash loop here.

Reply:
Notice the use of `xargs` here. But I see your point and I may have a better way of doing this.